pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -236,7 +241,27 @@ func (f *Fit) PreFilter(ctx context.Context, cycleState fwk.CycleState, pod *v1.
 		// and the older (before v1.28) kubelet, make the Pod unschedulable.
 		return nil, fwk.NewStatus(fwk.UnschedulableAndUnresolvable, "Pod has a restartable init container and the SidecarContainers feature is disabled")
 	}
-	cycleState.Write(preFilterStateKey, computePodResourceRequest(pod, ResourceRequestsOptions{EnablePodLevelResources: f.enablePodLevelResources}))
+	result := computePodResourceRequest(pod, ResourceRequestsOptions{EnablePodLevelResources: f.enablePodLevelResources})
+	if f.enableDRAExtendedResource && f.handle != nil {
+		draManager := f.handle.SharedDRAManager()
+		classes, err := draManager.DeviceClasses().List()
```

Comment:
That's the same code as in `dynamicresources.go`. Please add the shared package that I suggested earlier to avoid the code duplication. Also add unit tests for that code.

Your argument that this code has so little overhead that a more complicated scheme for only computing this once per scheduling cycle seems valid. But we should still use a shared implementation of the computation itself.

Reply:
sure, extracted it into extended package, that is shared / used by both noderesources and dynamicresources plugin